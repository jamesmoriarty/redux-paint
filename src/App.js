import React from "react";
import { connect } from "react-redux";
import { makeStyles } from "@material-ui/core/styles";
import AppBar from "@material-ui/core/AppBar";
import Toolbar from "@material-ui/core/Toolbar";
import Typography from "@material-ui/core/Typography";
import Card from "@material-ui/core/Card";
import Container from "@material-ui/core/Container";
import Canvas from "./components/Canvas";
import UndoButton from "./components/UndoButton";
import RedoButton from "./components/RedoButton";

const useStyles = makeStyles((theme) => ({
  title: {
    flexGrow: 1,
  },
  menuButton: {
    marginRight: theme.spacing(2),
  },
  content: {
    flexGrow: 1,
    height: "100vh",
    overflow: "auto",
  },
  container: {
    paddingTop: theme.spacing(8),
    paddingBottom: theme.spacing(8),
  },
  canvas: {
    height: "70vh",
  },
}));

function App({ lines, dispatch }) {
  const classes = useStyles();

  return (
    <div>
      <AppBar position="relative">
        <Toolbar>
          <Typography variant="h6" className={classes.title}>
            Redux Paint
          </Typography>
          <UndoButton className={classes.menuButton}/>
          <RedoButton className={classes.menuButton}/>
        </Toolbar>
      </AppBar>
      <main className={classes.content}>
        <Container maxWidth="lg" className={classes.container}>
          <Card className={classes.canvas} raised={true}>
            <Canvas />
          </Card>
        </Container>
      </main>
    </div>
  );
}

const mapStateToProps = (state) => ({
  lines: state.lines,
  future: state.future,
});

export default connect(mapStateToProps, null)(App);
