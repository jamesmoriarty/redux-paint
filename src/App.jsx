import React from "react";
import { connect } from "react-redux";
import { makeStyles } from "@material-ui/core/styles";
import AppBar from "@material-ui/core/AppBar";
import Toolbar from "@material-ui/core/Toolbar";
import Typography from "@material-ui/core/Typography";
import Card from "@material-ui/core/Card";
import Container from "@material-ui/core/Container";
import Button from "@material-ui/core/Button";
import UndoIcon from "@material-ui/icons/Undo";
import RedoIcon from "@material-ui/icons/Redo";
import GestureIcon from "@material-ui/icons/Gesture";
import LineIcon from "@material-ui/icons/Remove";
import RectIcon from "@material-ui/icons/CheckBoxOutlineBlank";
import ColorIcon from "@material-ui/icons/Palette";
import ButtonGroup from "@material-ui/core/ButtonGroup";
import Canvas from "./components/Canvas";
import ColorPicker from "./components/ColorPicker";
import { OP_TYPES, ACTION_TYPES } from "./constants";

const useStyles = makeStyles((theme) => ({
  title: {
    flexGrow: 1,
  },
  toolButtonGroup: {
    marginRight: theme.spacing(2),
  },
  content: {
    flexGrow: 1,
    height: "100vh",
    overflow: "auto",
  },
  container: {
    paddingTop: theme.spacing(12),
    paddingBottom: theme.spacing(8),
  },
  canvas: {
    height: "70vh",
  },
}));

function App({ history, future, type, dispatch, color }) {
  const classes = useStyles();

  return (
    <div>
      <AppBar position="relative">
        <Toolbar>
          <Typography variant="h6" className={classes.title}>
            Redux Paint
          </Typography>
          <ButtonGroup className={classes.toolButtonGroup}>
            <ColorPicker>
              <Button variant="contained">
                <ColorIcon style={{ color: color }} />
              </Button>
            </ColorPicker>
          </ButtonGroup>
          <ButtonGroup className={classes.toolButtonGroup}>
            <Button
              variant="contained"
              color={
                type === OP_TYPES.OP_TYPE_GESTURE ? "secondary" : "default"
              }
              onClick={() =>
                dispatch({
                  type: ACTION_TYPES.OP_SET_TYPE,
                  payload: { type: OP_TYPES.OP_TYPE_GESTURE },
                })
              }
            >
              <GestureIcon />
            </Button>
            <Button variant="contained" disabled={true}>
              <LineIcon />
            </Button>
            <Button
              variant="contained"
              color={type === OP_TYPES.OP_TYPE_RECT ? "secondary" : "default"}
              onClick={() =>
                dispatch({
                  type: ACTION_TYPES.OP_SET_TYPE,
                  payload: { type: OP_TYPES.OP_TYPE_RECT },
                })
              }
            >
              <RectIcon />
            </Button>
          </ButtonGroup>
          <ButtonGroup className={classes.toolButtonGroup}>
            <Button
              variant="contained"
              disabled={history.length === 0}
              onClick={() => {
                dispatch({ type: ACTION_TYPES.UNDO, payload: {} });
              }}
            >
              <UndoIcon />
            </Button>
            <Button
              variant="contained"
              disabled={future.length === 0}
              onClick={() => {
                dispatch({ type: ACTION_TYPES.REDO, payload: {} });
              }}
            >
              <RedoIcon />
            </Button>
          </ButtonGroup>
        </Toolbar>
      </AppBar>
      <main className={classes.content}>
        <Container className={classes.container}>
          <Card className={classes.canvas} raised={true}>
            <Canvas />
          </Card>
        </Container>
      </main>
    </div>
  );
}

const mapStateToProps = (state) => state;

export default connect(mapStateToProps, null)(App);
