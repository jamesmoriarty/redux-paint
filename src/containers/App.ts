import App from "../App";
import { ACTION_TYPES, OP_TYPES, State, Dispatch } from "../types";
import { connect } from "react-redux";

const mapStateToProps = (state: State) => state;

const mapDispatchToProps = (dispatch: Dispatch) => {
  return {
    onGesture: () => {
      dispatch({
        type: ACTION_TYPES.OP_SET_TYPE,
        payload: { type: OP_TYPES.OP_TYPE_GESTURE },
      });
    },
    onRect: () => {
      dispatch({
        type: ACTION_TYPES.OP_SET_TYPE,
        payload: { type: OP_TYPES.OP_TYPE_RECT },
      });
    },
    onUndo: () => {
      dispatch({ type: ACTION_TYPES.UNDO, payload: {} });
    },
    onRedo: () => {
      dispatch({ type: ACTION_TYPES.REDO, payload: {} });
    },
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(App);
