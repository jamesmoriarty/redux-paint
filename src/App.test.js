import React from "react";
import { act, render } from "@testing-library/react";
import App from "./App";

test("renders html", () => {
  act(() => {
    expect(render(<App />)).toMatchSnapshot();
  });
});
