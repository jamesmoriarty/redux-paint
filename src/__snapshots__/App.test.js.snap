// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders html 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <header
          class="MuiPaper-root MuiAppBar-root MuiAppBar-positionRelative MuiAppBar-colorPrimary MuiPaper-elevation4"
        >
          <div
            class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
          >
            <h6
              class="MuiTypography-root makeStyles-title-1 MuiTypography-h6"
            >
              Redux Paint
            </h6>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained Mui-disabled Mui-disabled"
              disabled=""
              tabindex="-1"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                Undo
              </span>
            </button>
          </div>
        </header>
        <main
          class="makeStyles-content-2"
        >
          <div
            class="MuiContainer-root makeStyles-container-3 MuiContainer-maxWidthLg"
          >
            <div
              class="MuiPaper-root MuiCard-root makeStyles-canvas-4 MuiPaper-elevation8 MuiPaper-rounded"
            >
              <canvas />
            </div>
          </div>
        </main>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <header
        class="MuiPaper-root MuiAppBar-root MuiAppBar-positionRelative MuiAppBar-colorPrimary MuiPaper-elevation4"
      >
        <div
          class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
        >
          <h6
            class="MuiTypography-root makeStyles-title-1 MuiTypography-h6"
          >
            Redux Paint
          </h6>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained Mui-disabled Mui-disabled"
            disabled=""
            tabindex="-1"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              Undo
            </span>
          </button>
        </div>
      </header>
      <main
        class="makeStyles-content-2"
      >
        <div
          class="MuiContainer-root makeStyles-container-3 MuiContainer-maxWidthLg"
        >
          <div
            class="MuiPaper-root MuiCard-root makeStyles-canvas-4 MuiPaper-elevation8 MuiPaper-rounded"
          >
            <canvas />
          </div>
        </div>
      </main>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
